src/instructlab/cli/model/evaluate.py

```diff
@@ -0,0 +1,540 @@
+    cls=clickext.ConfigOption,
+    config_sections="mmlu_branch",
+)
+@click.option(
+    "--gpus",
+    type=click.IntRange(min=0),
+    help="Number of GPUs to utilize for evaluation (not applicable to llama-cpp)",
+)
+@click.option(
+    "--merge-system-user-message",
+    is_flag=True,
+    help="Indicates whether to merge system and user message for mt_bench and mt_bench_branch (required for Mistral based judges)",
+)
+@click.option(
+    "--backend",
+    type=click.Choice(tuple(backends.SUPPORTED_BACKENDS)),
+    help="Serving backend to use for the model and base model (if applicable) during evaluation. Options are vllm and llama-cpp.",
+)
+@click.option(
+    "--judge-backend",
+    type=click.Choice(tuple(backends.SUPPORTED_BACKENDS)),
+    help="Serving backend to use for the judge model for during mt_bench or mt_bench_branch evaluation. Options are vllm and llama-cpp.",
+)
+@click.option(
+    "--tls-insecure",
+    is_flag=True,
+    help="Disable TLS verification for model serving.",
+)
+@click.option(
+    "--tls-client-cert",
+    type=click.Path(),
+    default="",
+    show_default=True,
+    help="Path to the TLS client certificate to use for model serving.",
+)
+@click.option(
+    "--tls-client-key",
+    type=click.Path(),
+    default="",
+    show_default=True,
+    help="Path to the TLS client key to use for model serving.",
+)
+@click.option(
+    "--tls-client-passwd",
+    type=click.STRING,
+    default="",
+    help="TLS client certificate password for model serving.",
+)
+@click.option(
+    "--enable-serving-output",
+    is_flag=True,
+    help="Print serving engine logs.",
+)
+@click.pass_context
+@clickext.display_params
+def evaluate(
+    ctx,
+    model,
+    base_model,
+    benchmark,
+    judge_model,
+    output_dir,
+    max_workers: str | int,
+    taxonomy_path,
+    branch,
+    base_branch,
+    few_shots,
+    batch_size: str | int,
+    tasks_dir,
+    gpus,
+    merge_system_user_message,
+    backend,
+    judge_backend,
+    tls_insecure,  # pylint: disable=unused-argument
+    tls_client_cert,  # pylint: disable=unused-argument
+    tls_client_key,  # pylint: disable=unused-argument
+    tls_client_passwd,  # pylint: disable=unused-argument
+    enable_serving_output,
+):
+    """Evaluates a trained model"""
+
+    # First Party
+    from instructlab.eval.exceptions import EvalError
+
+    with contextlib.suppress(ValueError):
+        max_workers = int(max_workers)
+    with contextlib.suppress(ValueError):
+        batch_size = int(batch_size)
+
+    try:
+        # get appropriate evaluator class from Eval lib
+        validate_options(
+            model,
+            base_model,
+            benchmark,
+            judge_model,
+            output_dir,
+            max_workers,
+            taxonomy_path,
+            branch,
+            base_branch,
+            few_shots,
+            batch_size,
+            tasks_dir,
+        )
+
+        if benchmark == Benchmark.MT_BENCH:
```

Comment:
I feel like all of this should not be in the CLI package, right? This seems to be functionality we would want on a lower level so it can be wrapped and called by an API. This goes for all benchmarks

Reply:
Moved em over :+1: